cmake/OpenCVUtils.cmake

```diff
@@ -810,3 +785,35 @@ function(ocv_add_library target)
 
   _ocv_append_target_includes(${target})
 endfunction()
+
+# build the list of opencv libs and dependencies for all modules
+#  _modules - variable to hold list of all modules
+#  _extra - variable to hold list of extra dependencies
+#  _3rdparty - variable to hold list of prebuilt 3rdparty libraries
+macro(ocv_get_all_libs _modules _extra _3rdparty)
+  set(${_modules} "")
+  set(${_extra} "")
+  set(${_3rdparty} "")
+  foreach(m ${OPENCV_MODULES_PUBLIC})
+    get_target_property(deps ${m} INTERFACE_LINK_LIBRARIES)
+    list(INSERT ${_modules} 0 ${deps} ${m})
+    foreach (dep ${deps} ${OPENCV_LINKER_LIBS})
+      if (TARGET ${dep})
```

Comment:
@alalek, if you mean something like:
```.cmake
foreach(m ${OPENCV_MODULES_PUBLIC})
  get_target_property(deps ${m} INTERFACE_LINK_LIBRARIES)
  list(INSERT ${_modules} 0 ${m})
  foreach (dep ${deps} ${OPENCV_LINKER_LIBS})
    if (DEFINED OPENCV_MODULE_${dep}_LOCATION)
      list(INSERT ${_modules} 0 ${dep})
    elseif (TARGET ${dep})
      list(INSERT ${_3rdparty} 0 ${dep})
    else()
      list(INSERT ${_extra} 0 ${dep})
    endif()
  endforeach()
endforeach()
```
I think it will work.